verifier.go

```diff
@@ -0,0 +1,99 @@
+package httpsignatures
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"fmt"
+	"strings"
+)
+
+// NewVerifier describes a function for creating a new instance of Verifier.
+type NewVerifier func(publicKey crypto.PublicKey, cryptoHash crypto.Hash) (Verifier, error)
+
+// newVerifier fetches the correct Verifier from the given signingAlgo. If an algo is provided,
+// it will be verified against the hashing algorithm in signingAlgo.
+func newVerifier(publicKey crypto.PublicKey, algo crypto.Hash, signingAlgo string) (Verifier, error) {
+	var signingHash string
+	var verifier NewVerifier
+	if strings.HasPrefix(signingAlgo, rsaName) {
+		signingHash = strings.TrimPrefix(signingAlgo, rsaName+"-")
+		verifier = NewRSAVerifier
+	}
+
+	if algo != 0 {
+		cryptoHash, err := fetchCryptoHash(algo)
+		if err != nil {
+			return nil, NewInitialisationError("required hashing algorithm is unknown or unsupported", err)
+		}
+
+		if cryptoHash.Name != signingHash {
+			return nil, NewValidationError(
+				fmt.Sprintf(
+					"hashing algorithm provided in signature (%s) did not match expected algorithm (%s)",
+					signingHash,
+					cryptoHash.Name,
+				),
+			)
+		}
+	}
+
+	if verifier == nil {
+		return nil, NewValidationError(fmt.Sprintf("no verifier could be found for %s", signingAlgo))
+	}
+	return verifier(publicKey, getAlgoByName(signingHash))
+}
+
+// Verifier interface describes something that can verify signatures.
```

Comment:
```suggestion
// Verifier is the interface that wraps the basic Verify method.

Verify verifies if the signature is valid for the provided content. It returns error if the signature is invalid or any error encountered that caused the verify to stop early. Verify must not modify the slice data, even temporarily.

Implementations must not retain content.
```

I used [io.Writer](https://pkg.go.dev/io#Writer) docs as a template.